### ISSUE TITLE
False positives for number-leading-zero and rule-trailing-semicolon ?

### ISSUE THREAD

**Issue by benfrain** [stylelint/stylelint#267]
I have these two set in my config:

````
"number-leading-zero": [2, "never"],
"rule-trailing-semicolon": [2, "always"],
````

However both are reporting as finding offending rules but there are none. For example a search for "0." reveals no matches in the CSS file (which is what I would expect number-leading-zero to find)

There are also no lines that end without a semi-colon. Am I misconfiguring?

**Comment by davidtheclark** [comment]
Hm ... bummer ... I'm not sure how to troubleshoot this. Is there any way you could provide more information so I could reproduce the problem?

**Comment by benfrain** [comment]
Hi David, I'll make a gist with my CSS file and config later. Thanks. Likelihood is it's something I'm doing wrong!

**Comment by benfrain** [comment]
Hi @davidtheclark I've created a Gist here, I included the output I get as text, gulpfile.js and package.json  along with the CSS I was running it on. Minification and Autoprefixer were disabled so this is the actual content stylelint was running against. https://gist.github.com/benfrain/d89f73bf50ac95c5825a

**Comment by davidtheclark** [comment]
Thanks. What's weird to me about this is that when I look at `test.css` I see that those warnings are actually *correct*: there *are* leading zeroes, and there *are* missing trailing semicolons.

I'm assuming that the `test.css` file you included is the CSS *after* it's been processed by the plugins preceding `stylelint` in your processor list. Is that right? So if the actual source files you're starting with do not have leading zeroes and missing trailing semicolons but the processed file you included does, that must mean that one of those preceding processors is changing your source ---- could that be right?

**Comment by davidtheclark** [comment]
Maybe we could isolate the problem better if we focus one one rule set that is trigger both warnings. In `test.css` I see

```css
.hd-Nav_Wrapper {
	display: flex;
	position: fixed;
	top: 0;
	width: 100%;
	padding-left: 30%;
	padding-top: 22px;
	min-height: 150vh;
	background-color: #333;
	transform: translate3d(-130%, 0, 0);
	backface-visibility: hidden;
	will-change: position;
	flex-direction: column;
	transition: transform .5s cubic-bezier(0.68, -0.55, 0.265, 1.55);
	z-index: 10
}
```

That has leading zeroes and lacks a trailing semicolon.

Could you identify where in your source files that rule set comes from, and we can see what transformations might have caused the confusion?

Another thing to try also would be to move `stylelint` up in your processing order. How about trying importing, then nesting, *then linting*, then everything else?

**Comment by benfrain** [comment]
Thanks, I think the most sensible thing for me to do now the nesting has been addressed (amazing btw!) is to go back to having the two separate tasks again. That way I can use a smaller subset of plugins. I'll update later once I've tried that. Thanks for all the assistance.

**Comment by benfrain** [comment]
@davidtheclark Just to conclude, switching back to the separate tasks is definitely the best approach for me. Once I pull down the new version with the nesting fix I think everything I'm currently attempting to lint will work sweet. Great project. Great help. :+1:

**Comment by davidtheclark** [comment]
Fantastic.

We worked on fixing nesting for a couple of rules -- please don't hesitate to report more issues as you run into problems: we'd like to fix them promptly.
